client/blocks/login/index.jsx

```diff
@@ -137,13 +137,17 @@ class Login extends Component {
 		if ( this.props.onTwoFactorRequested ) {
 			this.props.onTwoFactorRequested( authType );
 		} else {
+			const currentURLQueryParameters = this.props?.redirectTo
+				? Object.fromEntries( new URL( this.props.redirectTo ).searchParams.entries() )
```

Comment:
We need [something like this](https://github.com/Automattic/wp-calypso/blob/888ab78745549c6af6f966a58875fd14d9aea3d5/packages/i18n-utils/src/localize-url.tsx#L162-L166) here. Some `redirectTo` urls are relative urls and will throw as errors:

```
 new URL('/example'); // throws
```

Reply:
I did some tests with relative and absolute paths and I have no errors. Do you think we have to use this try/catch anyway?

Reply:
You probably need to add some logging, I mentioned [this code](https://github.com/Automattic/wp-calypso/blob/00429b19b070d7d39f8208ac5db1a7d27e92d43f/client/blocks/login/login-form.jsx#L193) in the post mortem. It's silently catching and logging the error to tracks.

e.g. http://calypso.localhost:3000/log-in?redirect_to=/home looks like this at the 2fa step (what the breakage looks like - no where to enter in your 2fa code)

![Screenshot 2022-07-12 at 16-13-07 Log In — WordPress com](https://user-images.githubusercontent.com/811776/178421213-16df8891-f043-4d3e-a76a-03d5bd2f8119.png)

If you add a `console.log(error)` in that catch you'll see this error:
```
 TypeError: URL constructor: /home is not a valid URL.
```